projects/valtimo/components/src/lib/components/left-sidebar/left-sidebar.component.ts

```diff
@@ -98,19 +99,41 @@ export class LeftSidebarComponent implements AfterViewInit, OnDestroy {
   navigateToRoute(route: Array<string>, event: MouseEvent) {
     event.preventDefault();
 
-    this.router.navigate(route);
+    if (!event.ctrlKey && !event.metaKey) {
+      this.router.navigate(route);
+
+      combineLatest([
+        this.shellService.sideBarExpanded$,
+        this.shellService.largeScreen$,
+        this.shellService.collapsibleWidescreenMenu$,
+      ])
+        .pipe(take(1))
+        .subscribe(([sideBarExpanded, largeScreen, collapsibleWidescreenMenu]) => {
+          if ((!largeScreen || collapsibleWidescreenMenu) && sideBarExpanded) {
+            this.shellService.setSideBarExpanded(false);
+          }
+        });
+    }
+  }
 
-    combineLatest([
-      this.shellService.sideBarExpanded$,
-      this.shellService.largeScreen$,
-      this.shellService.collapsibleWidescreenMenu$,
-    ])
-      .pipe(take(1))
-      .subscribe(([sideBarExpanded, largeScreen, collapsibleWidescreenMenu]) => {
-        if ((!largeScreen || collapsibleWidescreenMenu) && sideBarExpanded) {
-          this.shellService.setSideBarExpanded(false);
-        }
-      });
+  onRightClick(sequence: string): boolean {
+    this.overflowMenuSequence$.next(sequence);
+
+    return false;
+  }
+
+  onOverflowMenuClosed(sequence: string): void {
+    this.overflowMenuSequence$.pipe(take(1)).subscribe(overflowMenuSequence => {
+      if (overflowMenuSequence === sequence) {
+        this.overflowMenuSequence$.next('');
+      }
+    });
```

Comment:
Will it work if you add this condition in a filter operator to avoid the conditional in the subscribe?